### ISSUE TITLE
Fix documentation to explain how schedules work in optimizers.

### ISSUE THREAD

**Issue by rosshemsley** [google-deepmind/optax#399]
### Discussed in https://github.com/deepmind/optax/discussions/390

<div type='discussions-op-text'>

<sup>Originally posted by **nalzok** August 13, 2022</sup>
For example, in the documentation for [`optax.adam`](https://optax.readthedocs.io/en/latest/api.html?highlight=adam#optax.adam), we have

> learning_rate (`Union[float, Callable[[Union[ndarray, float, int]], Union[ndarray, float, int]]]`) – this is a fixed global scaling factor.

I noticed that `learning_rate` can be a function. How can a function be "a fixed global scaling factor"? What parameters should such a function take, and how does its return value affect the optimization process given that Adam has a learning rate schedule on its own?</div>

**Comment by vroulet** [comment]
Done in #778
